### DISCUSSION TITLE
How to remove GitHub information (stars, forks, ...)?

### DISCUSSION THREAD

**Discussion by do-me** [squidfunk/mkdocs-material#4889] [Q&A] [answered]
I just noted, that depending on the screen width, in the mobile layout the sidebar sometime does / doesn't contain the GitHub banner element. The GitHub element is only shown on the narrowest screen width. 
![image](https://user-images.githubusercontent.com/47481567/213643824-7b573d52-afba-4308-9ebd-2849e4812e9a.png)

Is there some logic behind it? I would assume the less space I have on a screen the less to show. Also, I doubt that people will get heavily involved in GitHub on their phones anyway. 

I hope I didn't miss an option to fully turn it off/on. If I didn't, I would propose just this: instead of the actual behavior I think it would an improvement to allow for on option in the mkdocs.yml to always hide or always show the element.

Reproducible on the current [mkdocs-material homepage](https://squidfunk.github.io/mkdocs-material).

**Comment by squidfunk** [comment]
The git information is always visible – it is kept in the header until the tablet portrait breakpoint, and then moved into the sidebar. You probably missed the point where it's still in the header. The displaying/switching logic is implemented with CSS, for example, here's the code that triggers the display of the git information in the header:

https://github.com/squidfunk/mkdocs-material/blob/5a228db63f7ce8936e220a78b261543f239b4ba7/src/assets/stylesheets/main/components/_header.scss#L245-L261

The reason is that there's not enough space to keep it in the header. Here's a video demonstrating the point at which it switches:

https://user-images.githubusercontent.com/932156/213654472-3bca2dc8-db42-4d19-9832-e5ae2dae0fb8.mp4

**Reply by squidfunk** [reply]
The behavior can be turned off by removing [`repo_url`](https://squidfunk.github.io/mkdocs-material/setup/adding-a-git-repository/#repository), or by [using some CSS to hide the git information altogether](https://github.com/squidfunk/mkdocs-material/discussions/2705#discussioncomment-796803).

**Comment by do-me** [comment, marked as answer]
Thanks for your reply. You are right, I was confusing myself as I already used custom CSS to hide the GitHub element on desktop sized windows. 
Due to that, it didn't make sense to me why suddenly the repo info appeared in the sidebar only on the smallest window width. 
I will probably just edit the partials as you described. 

However, I'd like to point out that there are (e.g. business) use cases where you want to keep a repo actively linked and the edit buttons enabled (for convenient editing) but you do not want to show the repo info to your clients (unnecessary). 

I suppose that this may be quite a common use case to get the best out of two worlds: 

1. Nice and secure static homepage & wiki for customers 
2. Convenient in-browser editing for (non-IT) colleagues. 

Considering this, what do you think about splitting up the repo option in: 
- enabling a repo and 
- optionally showing/hiding the repo info?

I guess that would be pretty much in line with the recent updates from version 8 to 9 where settings are much more granular.

**Reply by squidfunk** [reply]
We might re-evaluate when we release the next major version, but not something we'd consider for now. As mentioned, you can use CSS or edit the partials to hide the information. All you need to do is to override the [`source.html`](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/source.html) partial with an empty one, by using [theme extension](https://squidfunk.github.io/mkdocs-material/customization/#setup-and-theme-structure). That's **one additional line of configuration** and **one empty file**. In `mkdocs.yml`:

``` yaml
theme:
  name: material
  custom_dir: overrides
```

Project structure:

``` sh
.
├─ overrides/
│  └─ partials/
│     └─ source.html # <-- Empty file
└─ mkdocs.yml
```

**Reply by do-me** [reply]
Just for completeness: I went the CSS way and added this line to my custom CSS. 
```css
.md-nav__source{display: none;}
```

**Reply by squidfunk** [reply]
Perfect, thanks for sharing. Note that hiding through CSS will still make the requests against GitHub's API, but at least it isn't visible. Overriding the partial will eradicate the requests completely. This is just meant as an FYI to future readers.

**Reply by do-me** [reply]
That makes sense, thanks for the note. Then I'll better stick to the partial solution.
